src/components/PositionsList/PositionsList.tsx

```diff
@@ -43,8 +47,14 @@ export const PositionsList: React.FC<IProps> = ({
   const navigate = useNavigate()
   const [defaultPage] = useState(initialPage)
   const [page, setPage] = useState(initialPage)
+  const dispatch = useDispatch()
+  const { length, loadedPages } = useSelector(positionsList)
 
   const handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (Object.keys(loadedPages).length * POSITIONS_PER_QUERY < Number(length)) {
+      dispatch(actions.getRemainingPositions())
```

Comment:
it's better to move dispatch from component to container.

Reply:
I totally agree